exo/networking/manual/network_topology_config.py

```diff
@@ -0,0 +1,48 @@
+from typing import Dict
+from dataclasses import dataclass
+
+import json
+
+from exo.topology.device_capabilities import DeviceCapabilities
+
+
+@dataclass
+class PeerConfig:
+  address: str
+  port: int
+  device_capabilities: DeviceCapabilities
+
+
+@dataclass
+class NetworkTopology:
+  """Configuration of the network. A collection outlining all nodes in the network, including the node this is running from."""
+
+  peers: Dict[str, PeerConfig]
+  """
+  node_id to PeerConfig. The node_id is used to identify the peer in the discovery process. The node that this is running from should be included in this dict.
+  """
+
+  @classmethod
+  def from_path(cls, path: str) -> "NetworkTopology":
```

Comment:
This could be simplified drastically if you're ok with including an additional dependency on pydantic https://docs.pydantic.dev/latest/examples/files/ (and we could get json + toml + yml for free)

Reply:
lets do this. fine with having dependency on `pydantic` for this